back/app/jobs/files/generate_preview_job.rb

```diff
@@ -0,0 +1,14 @@
+module Files
+  class GeneratePreviewJob < ::ApplicationJob
+    queue_as :default
+
+    def run(preview)
+      PreviewService.new.generate_preview_content(preview)
+    rescue StandardError => e
+      if error_count >= (self.class.maximum_retry_count - 1)
+        preview.update!(status: 'failed')
+      end
+      raise e
```

Comment:
(nitpicky) `raise e` resets the backtrace

```suggestion
    rescue StandardError
      if error_count >= (self.class.maximum_retry_count - 1)
        preview.update!(status: 'failed')
      end

      raise
```